sdk/servicebus/service-bus/src/core/linkEntity.ts

```diff
@@ -118,13 +123,31 @@ export class LinkEntity {
     await defaultLock.acquire(this._context.namespace.cbsSession.cbsLock, () => {
       return this._context.namespace.cbsSession.init();
     });
-    const tokenObject = await this._context.namespace.tokenProvider.getToken(this.audience);
+    let tokenObject: AccessToken;
+    let tokenType: TokenType;
+    if (this._context.namespace.tokenCredential instanceof SharedKeyCredential) {
+      tokenObject = this._context.namespace.tokenCredential.getToken(this.audience);
+      tokenType = TokenType.CbsTokenTypeSas;
+      // renew sas token in every 45 minutess
+      this._tokenTimeout = (3600 - 900) * 1000;
+    } else {
+      const aadToken = await this._context.namespace.tokenCredential.getToken(Constants.aadServiceBusScope);
+      if (!aadToken) {
+        throw new Error(`Failed to get token from the provided "TokenCredential" object`);
+      }
+      tokenObject = aadToken;
+      tokenType = TokenType.CbsTokenTypeJwt;
+      this._tokenTimeout = tokenObject.expiresOnTimestamp - Date.now() - 2 * 60 * 1000;
+    }
     log.link(
       "[%s] %s: calling negotiateClaim for audience '%s'.",
       this._context.namespace.connectionId,
       this._type,
       this.audience
     );
+    if (!tokenObject) {
+      throw new Error("Token cannot be null");
+    }
```

Comment:
The above if check is repeated a few lines below, probably due to merge conflict resolution